### PR TITLE
Feat: User signup with OTP verification

### PR DESCRIPTION
**Pull request by adelajaOlamilekan**
​
## Description
This PR removes the email validation and strong password check from the backend (they will be done at the frontend).
It adds an endpoint for uses to signup after their email has been verified through an otp.

​
## Related Issue (Link to linear ticket)

​
## Motivation and Context
The motivation for this PR comes from the need to handle data validation early on (i.e at frontend) and also confirm that users enter a correct mail

​
## How Has This Been Tested?
This endpoint has been tested using POSTMAN to ensure all cases are handled properly and also checking if an email is received by the user.

​
## Screenshots (if appropriate - Postman, etc):
Screenshot (1325)
Screenshot (1326)
Screenshot (1327)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
​
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/routes/auth_routes.py [hunk 1]
```diff
@@ -40,6 +42,43 @@
...
+        UserResponse: The response object.
+    """
+
+    # check if user exists
+    requested_user = db.query(User).filter_by(username=user.username.lower()).first()
+
+    if requested_user:
+        raise HTTPException(status_code=404, detail="Username exists already.")
+    
+    #generate otp
```
Comment (zxenonx) [inline app/routes/auth_routes.py:69]: this comment is not necessary

File: app/routes/auth_routes.py [hunk 2]
```diff
@@ -40,6 +42,43 @@
...
+    otp = get_otp()
+
+    #send otp to mail
```
Comment (zxenonx) [inline app/routes/auth_routes.py:72]: this one is also not necessary.

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @adelajaOlamilekan 
I left some comments.

File: app/routes/auth_routes.py (hunk 1 above)
Comment (adelajaOlamilekan) [reply app/routes/auth_routes.py:69]: Done

File: app/routes/auth_routes.py (hunk 2 above)
Comment (adelajaOlamilekan) [reply app/routes/auth_routes.py:72]: Done

File: app/services/mail_service.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def send_video(username: str, video_id: str, recipient_address: str):
         smtp.send_message(msg)
 
 
-def send_otp(recipient_address: str, otp: str):
+def send_otp(recipient_address: str, otp: str, subject: str):
```
[the remark this answers is not in the document]
Comment (adelajaOlamilekan) [reply app/services/mail_service.py:55]: Resolved

**Review by zxenonx — APPROVED** [review]
Good.

**Pull request merged**